ted/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -634,6 +654,20 @@ public void zeroRobotPosition()
     this.setRobotPosition(new Pose2d(0,0,Rotation2d.fromDegrees(0)));
   }
 
+  /**
+   * A method that updates the robot position with a vision measurement
+   * @param visionMeasurement the most recent vision measurement provided by vision subsystem
+   */
+  private void addVisionMeasurement(VisionMeasurement visionMeasurement){
+    // The wpilib matrix constructor requires sizes specified as Nat types. 
+    // trying to update the matrix dynamicaly leads to issues
+    Matrix<N3,N1> visionStdDev = (new MatBuilder<N3,N1>(Nat.N3(), Nat.N1())).fill(new double[]{0.9, 0.9, 0.9});
+    // for now ignore all vision measurements that are null or contained robot position is null
+    if (visionMeasurement != null && visionMeasurement.getRobotPosition() != null && useVision){
```

Comment:
Is the cost of visionMeasurement.getRobotPosition() low?  Is the memory cost of what it returns high?  Is it possible the state changes between successive calls into visionMeasurement.getRobotPosition()? Your code assumes the compute cost of the method is very low, the memory cost might be high-ish and that successive recompute of getRobotPosition will return the same result.

I'd bet that the above are less likely to be true (but then again, I'm not totally sure either).  Usually if I'm using a variable 2x from a method in quick succession, I decide to pay for the memory, save the compute and therefor gain consistency between usages. 

In essence, I'd imagine that I'd use code that looks like this instead:

    // for now ignore all vision measurements that are null or contained robot position is null
    if (visionMeasurement != null && useVision){
      Pose2d visionComputedMeasurement = visionMeasurement.getRobotPosition();
      if(visionComputedMeasurement != null) {
        swervePoseEstimator.addVisionMeasurement(visionMeasurement.getRobotPosition(), visionMeasurement.getTimestamp(), visionStdDev);
      }
    }

Reply:
resolved